### ISSUE TITLE
Permission denied for sed -i

### ISSUE THREAD

**Issue by an-kumar** [bazelbuild/rules_foreign_cc#306]
rules_foreign_cc uses `sed -i` as part of it's workflow; but bazel creates all output directories with mode 0555, which is not writeable. Since `sed -i` needs write permissions to the directory (to make a temporary file), sed commands fail with permission denied.

What's the work-around for this?

**Comment by mattgodbolt** [comment]
I just hit this too, and it seems to prevent things like the substituion of `BAZEL_GEN_ROOT` from working:

```
_____ BEGIN BUILD LOGS _____
Bazel external C/C++ Rules #0.0.8. Building library 'curl'
Environment:______________BUILD_SCRIPT=bazel-out/k8-fastbuild/bin/curl/logs/Configure_script.sh
EXT_BUILD_ROOT=/home/mgodbolt/.cache/bazel/_bazel_mgodbolt/d5bad8eff3da01980a0963475d1edfc1/execroot/__main__
BUILD_LOG=bazel-out/k8-fastbuild/bin/curl/logs/Configure.log
PWD=/home/mgodbolt/.cache/bazel/_bazel_mgodbolt/d5bad8eff3da01980a0963475d1edfc1/execroot/__main__
TMPDIR=/tmp
EXT_BUILD_DEPS=/home/mgodbolt/.cache/bazel/_bazel_mgodbolt/d5bad8eff3da01980a0963475d1edfc1/execroot/__main__/bazel_foreign_cc_deps_curl
BUILD_TMPDIR=/tmp/tmp.SrPYfRZLdf
SHLVL=2
INSTALLDIR=/home/mgodbolt/.cache/bazel/_bazel_mgodbolt/d5bad8eff3da01980a0963475d1edfc1/execroot/__main__/bazel-out/k8-fastbuild/bin/curl
PATH=/home/mgodbolt/.cache/bazel/_bazel_mgodbolt/d5bad8eff3da01980a0963475d1edfc1/execroot/__main__:/home/mgodbolt/.autojump/bin:/home/mgodbolt/bin:/home/mgodbolt/dev/personal/compiler-explorer/bin:home/mgodbolt/dev/coral/web/octopus/.node-12.4.0/bin:/home/mgodbolt/dev/coral/web/octopus/.node-12.4.0/bin:/home/mgodbolt/.cargo/bin:/home/mgodbolt/miniconda3/condabin:/home/mgodbolt/.autojump/bin:/usr/local/bin:/usr/local/sbin:/usr/bin:/usr/sbin
_=/usr/bin/env
__________________________sed: couldn't open temporary file /home/mgodbolt/.cache/bazel/_bazel_mgodbolt/d5bad8eff3da01980a0963475d1edfc1/execroot/__main__/bazel_foreign_cc_deps_curl/openssl/lib/pkgconfig/sed3n3D9M: Permission denied
sed: couldn't open temporary file /home/mgodbolt/.cache/bazel/_bazel_mgodbolt/d5bad8eff3da01980a0963475d1edfc1/execroot/__main__/bazel_foreign_cc_deps_curl/openssl/lib/pkgconfig/sedsB8C6M: Permission denied
sed: couldn't open temporary file /home/mgodbolt/.cache/bazel/_bazel_mgodbolt/d5bad8eff3da01980a0963475d1edfc1/execroot/__main__/bazel_foreign_cc_deps_curl/openssl/lib/pkgconfig/sed5RSoRM: Permission denied
PKG_CONFIG_PATH=:/home/mgodbolt/.cache/bazel/_bazel_mgodbolt/d5bad8eff3da01980a0963475d1edfc1/execroot/__main__/bazel-out/k8-fastbuild/bin/copy_openssl/openssl/lib/pkgconfig
checking whether to enable maintainer-specific portions of Makefiles... no
```

and then later:

```
  CC       vtls/libcurl_la-mbedtls.lo
  CC       vtls/libcurl_la-mesalink.lo
  CCLD     libcurl.la
../libtool: line 7566: cd: BAZEL_GEN_ROOT/openssl/lib: No such file or directory
libtool:   error: cannot determine absolute directory name of 'BAZEL_GEN_ROOT/openssl/lib'
```

where clearly the libtool stuff hasn't had its `BAZEL_GEN_ROOT` substituted :\

**Comment by irengrig** [comment]
Fixed by #324
